### ISSUE TITLE
Some problem in TargetAssigner

### ISSUE THREAD

**Issue by muzi2045** [jhultman/PV-RCNN#9]
I am trying to train my own lidar dataset(9 classes) on the repo.
And here seems something wrong in tensor shape mismatch which occured in ProposalTargetassigner part.
```
matches = torch.stack(matches).view(self.anchors.shape[:-1])
match_labels = torch.stack(match_labels).view(self.anchors.shape[:-1])
```
![捕获](https://user-images.githubusercontent.com/12984898/75852210-70a74500-5e26-11ea-9ec2-ee63025fe385.PNG)

here is my log:
![proposal_targets py - PV-RCNN  SSH_ udi tpddns cn  - Visual Studio Code 2020_3_4 14_33_36 (2)](https://user-images.githubusercontent.com/12984898/75851811-9253fc80-5e25-11ea-9470-a1b36a78bb58.png)

what's the meaning of full_idx in get_matches func?
@jhultman

**Comment by muzi2045** [comment]
in multi-class training , the target assigner code have some problem.
https://github.com/jhultman/PV-RCNN/blob/ca40082be65531c03dfda7353f6eb2d6d47ebb64/pvrcnn/core/proposal_targets.py#L68

because per frame can't contain all class obstacle, so the matches and matches_label can't be just view as [class_num, yaw_num, x_range, y_range].
```
matches = torch.stack(matches).view(self.anchors.shape[:-1])
match_labels = torch.stack(match_labels).view(self.anchors.shape[:-1])
```

**Comment by jhultman** [comment]
Hi, thanks for reporting this. I have just pushed a fix -- let me know if it works for you.

`full_idx` is an indexing trick which maps the masked ground truth boxes for class `i` back to the full set of ground truth boxes (containing all classes). Sorry the naming is poor. Here is an example which I hope explains it.

```
import torch
i = 2
class_idx = torch.LongTensor([0, 2, 1, 0, 2])
full_idx = torch.arange(class_idx.shape[0])
class_mask = class_idx == i
matches = torch.LongTensor([0, 1, 0])
matches = full_idx[class_mask][matches]
print(matches)
>>> tensor([1, 4, 1])
```

**Comment by muzi2045** [comment]
seems like the tensor are not in GPU
the full_idx tensor need to load in GPU, that can resolve this problem.
![捕获](https://user-images.githubusercontent.com/12984898/75888544-84729b80-5e66-11ea-832d-1b7b4d298347.PNG)
![test](https://user-images.githubusercontent.com/12984898/75890162-ef24d680-5e68-11ea-9d5e-9b3c9a148b74.PNG)

**Comment by muzi2045** [comment]
the newest repo code can works in my own dataset now.
thanks
